### PR TITLE
[HIP] Clean up shuffle reduction in MDRange and in Team

### PR DESCRIPTION
**Pull request by Rombur**
- Remove `UseShflReduction` in `Parallel_HIP_MDRange` since it's always false
- Use tags in `Parallel_HIP_Team` similar to what we do in `Parallel_HIP_Range`
- Fix a `FIXME_HIP`: there was a comment that two functions were similar, but they only have the following code in common 
```C++
	// Iterate this block through the league
    for (int league_rank = blockIdx.x; league_rank < int_league_size;
         league_rank += gridDim.x) {
      this->template exec_team<work_tag>(...);
```

### CODE REVIEW

File: core/src/HIP/Kokkos_HIP_Parallel_Team.hpp [hunk 1]
```diff
@@ -705,9 +724,11 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
       threadid = base_thread_id;
     }
 
-    run(Kokkos::Impl::if_c<UseShflReduction, DummyShflReductionType,
-                           DummySHMEMReductionType>::select(1, 1.0),
-        threadid);
+    using ReductionTag =
+        typename std::conditional<UseShflReduction, ShflReductionTag,
+                                  SHMEMReductionTag>::type;
```
Comment (dalg24) [inline core/src/HIP/Kokkos_HIP_Parallel_Team.hpp:729]: Prefer `std::conditional_t` now that we require C++14

File: core/src/HIP/Kokkos_HIP_Parallel_Team.hpp [hunk 2]
```diff
@@ -716,7 +737,7 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
     }
   }
 
-  __device__ inline void run(DummySHMEMReductionType const&,
+  __device__ inline void run(SHMEMReductionTag const&,
```
Comment (dalg24) [inline core/src/HIP/Kokkos_HIP_Parallel_Team.hpp:740]: Consider taking by value

**Review by dalg24 — APPROVED** [review]

**Review by DavidPoliakoff — APPROVED** [review]
Looks sane, and nets us negative lines, 100% in favor

**Review by masterleinad — APPROVED** [review]
Looks OK. We could consider doing the corresponding changes to CUDA.

**Review by dalg24 — APPROVED** [review]

**Pull request merged**